website/source/docs/providers/aws/index.html.markdown

```diff
@@ -30,6 +30,23 @@ resource "aws_instance" "web" {
 }
 ```
 
+## Authentication 
+
+The AWS provider offers flexible means of providing credentials for
+authentication. Included is support including hard coded credentials, 
+environment variables, and shared credential files, in that order of precedence. 
+
+Terraform will first attempt to use an `access_key` and `secret_key` provided in 
+the `provider` block (shown in the example above). If those are omitted, it will 
+attempt to discover those values by referencing the `AWS_ACCESS_KEY_ID` and 
+`AWS_SECRET_ACCESS_KEY` environment variables. Lastly, if those are not found 
+it will look for credentials in the default location for a credentials file, or 
+the file path specified in the `shared_credentials_file` attribute of the 
+`provider` block.
```

Comment:
The shared credentials path can be overridden with `AWS_SHARED_CREDENTIALS_FILE`

And the creds file can have multiple profiles, which can be selected via `profile`, which can also be set via `AWS_PROFILE`.

https://github.com/aws/aws-sdk-go/blob/87b1e60a50b09e4812dee560b33a238f67305804/aws/credentials/shared_credentials_provider.go#L25-L36

I wonder if this would look clearer as a number listed of "things we check in this order" than a prose paragraph?